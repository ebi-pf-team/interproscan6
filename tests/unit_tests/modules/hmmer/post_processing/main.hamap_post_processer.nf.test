nextflow_process {

    name "Test Process HAMAP_POST_PROCESSER"
    script "interproscan/modules/hmmer/post_processing/main.nf"
    process "HAMAP_POST_PROCESSER"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/hamap/hamap.tlb.out"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = [
                    "path-to-model-dir",
                    "flags-for-pfsearch"
                ]
                input[1] = file("meta/tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa")
                input[2] = file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/hamap/hamap.tlb.out")
                input[3] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
