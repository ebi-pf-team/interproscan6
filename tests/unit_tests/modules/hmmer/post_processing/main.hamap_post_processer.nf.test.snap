{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hamap_pfsearch_output:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T11:29:11.803442"
    }
}